inst/include/conditionalSampler.h

```diff
@@ -0,0 +1,673 @@
+            void SetDigitsPrint(int newDigitsPrint) {digitsPrint = newDigitsPrint;}
+            ///Returns the reference trajectory
+            std::vector<Space> GetReferenceTrajectory(void) const {return referenceTrajectory;}
+            ///Returns a constant reference to the reference trajectory
+            const std::vector<Space> & GetReferenceTrajectoryRefs(void) const {return referenceTrajectory;}
+            ///Returns the n'th element of the reference trajectory
+            Space GetReferenceValue(long n) const {return referenceTrajectory[n];}
+            ///Returns a constant reference to the n'th element of the reference trajectory
+            const Space & GetReferenceValueRefs(long n) const {return referenceTrajectory[n];}
+            ///Returns index of conditional reference Trajectory
+            const long GetReferenceValueIndex(long lTime) const {return(referenceTrajectoryIndices.at(lTime));
+            }
+            ///Sets the reference trajectory
+            void SetReferenceTrajectory(const std::vector<Space>& newReferenceTrajectory) {referenceTrajectory = newReferenceTrajectory;}
+            ///Sets the n'elment of the reference trajectory
+            void SetReferenceValue(const Space & newReferenceValue, long n) {referenceTrajectory[n] = newReferenceValue;}
+            ///Initialise the conditional sampler and its constituent particles. The reference trajectory used to set the first coordinate at T=0 is either from the last call to conditionalSampler.SetReferenceTrajectory(), or defaults to the trajectory passed to the constructor at the time the object is created.
+            void Initialise(void);
+            ///Initialise the sampler and its constituent particles.
+            void Initialise(const std::vector<Space>& newReferenceTrajectory);
+            ///Perform one conditional iteration of the simulation algorithm.
+            void Iterate(void);
+            ///Perform one conditional iteration of the simulation algorithm and return the resulting ess
+            double IterateEss(void);
+            ///Perform conditional iterations until the specified evolution time is reached
+            void IterateUntil(long lTerminate);
+            ///Resample the particle set using the specified resampling scheme specifically adjusted for conditional resampling
+            void conditionalResample(ResampleType::Enum lMode);
+            ///Move the reference particle by setting the coordinate to the reference value and re-weighting.
+            void MoveReferenceParticle(void);
+            ///Dump a specified particle to the specified output stream in a human readable form
+            std::ostream & StreamParticle(std::ostream & os, long n) const;
+            ///Dump the entire particle set to the specified output stream in a human readable form
+            std::ostream & StreamParticles(std::ostream & os, int digits) const;
```

Comment:
I'm not proposing that we change anything in this PR, but we should have consistent interfaces for `sampler` and `conditionalSampler` insofar as we can, so we should add an issue/todo item somewhere to make the printing interfaces consistent. Whether the number of digits to print belongs within an individual sampler object or more globally isn't immediately obvious to me and I don't have strong views, but if it's handled by the conditional case it should also be handled by the base case, I would think.

Reply:
Yes indeed! Added this to the to-do list of GSoC. We might want to make an overall To-Do in the discussion section not related to GSoC for future work that also includes past To-Dos (including or reference #31 bullet point 4 here makes sense).

Reply:
Thanks. It might well be a good idea to have a more structured lto do list somewhere.